### ISSUE TITLE
[Bug]: dear teachers, when i try to change the softmax fuction on gpt2_spu_torch_version, there is a question

### ISSUE THREAD

**Issue by zhangwaer** [secretflow/spu#753]
### Issue Type

Build/Install

### Modules Involved

Others

### Have you reproduced the bug with SPU HEAD?

Yes

### Have you searched existing issues?

Yes

### SPU Version

spu0.9.1b0

### OS Platform and Distribution

ubuntu18.04

### Python Version

python3.11.8

### Compiler Version

GCC11.2.0

### Current Behavior?

May I implement the Torch version of softmax modification according to the method of JAX below?  :
thank you very much!!!

def hack_softmax(
    x: Array,
    axis: Optional[Union[int, Tuple[int, ...]]] = -1,
    where: Optional[Array] = None,
    initial: Optional[Array] = None,
) -> Array:
    x_max = jnp.max(x, axis, where=where, initial=initial, keepdims=True)
    x = x - x_max

    # exp on large negative is clipped to zero
    b = x > -14
    nexp = jnp.exp(x)

    divisor = jnp.sum(nexp, axis, where=where, keepdims=True)

    return b * (nexp / divisor)

@contextmanager
def hack_softmax_context(msg: str, enabled: bool = False):
    if not enabled:
        yield
        return
    # hijack some target functions
    raw_softmax = jnn.softmax
    jnn.softmax = hack_softmax
    yield
    # recover back
    jnn.softmax = raw_softmax

### Standalone code to reproduce the issue

```Python
print("A bug")
```

**Comment by anakinxc** [comment]
Hi @zhangwaer 

Yes, you can definitely use the one for JAX as a reference and implement a new one for Torch.

**Comment by zhangwaer** [comment]
> Hi @zhangwaer
> 
> Yes, you can definitely use the one for JAX as a reference and implement a new one for Torch.

thank you very much!!
